scss/bitstyles/tools/_color.scss

```diff
@@ -0,0 +1,52 @@
+@use "sass:map";
+@use "sass:math";
+
+// Generate palette
+//
+// **Generates a Sass map of colors. Each color is a `name: color` pair, with the name being the percentage as a string (one for each of the percentages in the list you pass in), and the color being the mix-color mixed into the base-color to that percentage.
+//
+// @param {String} $base-color  Base colour into which the other color will be mixed
+//
+// @param {String} $mix-color  The color to mix into the base color at different percentages
+//
+// @param {list} $percentages  A list of decmial values representing the percentages at which the mix-color will be mixed into the base-color
+
+@function generate-palette($base-color, $mix-color, $percentages) {
+  $color-map: ();
+
+  @each $percentage in $percentages {
+    $color-map: map.merge($color-map, ('#{$percentage}': mix($mix-color, $base-color, 100% - percentage(math.div($percentage, 100)))));
+  }
+
+  @return $color-map;
+}
+
+// Palette
+//
+// **Returns requested colours from the global `$palette` map.**
+// e.g. `color: palette('text');`
+//
+// @param {String} $color  Base colour within the palette
+//
+// @param {String} $percent   Optional. Percentage of the color to mix with white. Defaults to `100%` (fuly the specified color, no white)
+//
+
+// stylelint-disable at-rule-no-unknown
+@function palette($color, $percent: '100') {
```

Comment:
hmmm, why do we actually pre-compute percentage values and can't do that math here instead? This would allow us to use odd percentage values (98.5%) and not rely on the preconfigured percentage steps.

Reply:
Having the preset percentages is a feature :) We’re providing a limited palette of colors, so that we don’t e.g. end up with 20 different colors for borders because they’re set manually every time. If you need to override the percentages, there’s the `$bitstyles-color-mix-percentages` variable, and if you need to specify exact colors instead of generated colors (because of branding for example), then the map can be overridden manually.

None of which means it wouldn’t be valid to switch to a model of generating colors on-demand instead of in advance, but it probably does mean there isn’t really an advantage 🤔 wdyt?